sklearn/decomposition/_pca.py

```diff
@@ -336,11 +336,11 @@ def fit(self, X, y=None):
 
         Parameters
         ----------
-        X : array-like, shape (n_samples, n_features)
+        X : ndarray of shape (n_samples, n_features)
```

Comment:
It should accept array-like?

Reply:
Given that the code uses .shape, I thought it could only take ndarrays and not lists

Reply:
you have to check the call to `_validate_data` where the conversion from X as a list to X as a numpy array.

Reply:
ah...did not see that thanks